### PR TITLE
Add vagrant support for pouch

### PR DESCRIPTION
**Pull request by manlge**
Ⅰ. Describe what this PR did
Add Vagrant support for pouch.

Ⅱ. Does this pull request fix one issue?
The commit doesn't fix any issue.

Ⅲ. Why don't you add test cases (unit test/integration test)?
The commit just have scriptfile and Documents.

Ⅳ. Describe how to verify it
Install Vagrant and virtualBox

On MacOS or Linux
export POUCH_BUILD=true
vagrant up

On Windows
set POUCH_BUILD=true
vagrant up

Ⅴ. Special notes for reviews

### CODE REVIEW

**Comment by manlge** [comment]
@fuweid 
hack/vagrant/ doesn't work

**Comment by manlge** [comment]
@fuweid The hack/vagrant doesn't work when vagrant up

==> pouch-dev-node: Cloning into '/root/.goenv'...
==> pouch-dev-node: Downloading go1.9.1.linux-amd64.tar.gz...
==> pouch-dev-node: -> https://storage.googleapis.com/golang/go1.9.1.linux-amd64.tar.gz
==> pouch-dev-node: error: failed to download Go Linux 64bit 1.9.1.tar.gz
==> pouch-dev-node: BUILD FAILED
==> pouch-dev-node:  (Ubuntu 16.04 using go-build 1.21.0)
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

hack/vagrant just for developer,  doesn't have quick start for users, So I think we can merge  committed files to pouch and remove hack/vagrant directory from pouch.

File: VAGRANT.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+## Requirements
+
+* Vagrant 1.9.x or newer
+* VirtuaBox
+
+## Get started
+
+```bash
+vagrant up
+vagrant ssh -c "sudo su -l"
```
Comment (AkihiroSuda) [inline VAGRANT.md]: sudo -i?

Comment (manlge) [reply VAGRANT.md]: sudo su -l (L)

Comment (AkihiroSuda) [reply VAGRANT.md]: `sudo -i` does same thing without nesting su in sudo

Comment (manlge) [reply VAGRANT.md]: I'm sorry, "sudo -i" is good idea. I'll change it to "sudo -i" next commit.

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      systemctl start pouch
+      echo "alias docker='pouch'" >> ~/.bashrc
+    SHELL
+
+    if ENV["POUCH_BUILD"] == "true"
+      pouch.vm.provision "shell", inline: <<-SHELL
+        # configring environments for pouch
+        GOPATH=/root/go
+        apt-get install -y --no-install-recommends build-essential
+        wget --progress=bar:force:noscroll https://dl.google.com/go/go1.10.3.linux-amd64.tar.gz -O /tmp/go1.10.3.linux-amd64.tar.gz
```
Comment (fuweid) [inline Vagrantfile]: we use go1.9.1 to build binary right now.

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        tar xf /tmp/go1.10.3.linux-amd64.tar.gz -C /opt/
+        echo "export GOROOT=/opt/go" >> ~/.bashrc
+        echo "export GOPATH=$GOPATH" >> ~/.bashrc
+        cd /usr/bin && find /opt/go/bin -type f | xargs -n1 ln -s
+
+        mkdir -p $GOPATH/src/github.com/alibaba
+        ln -s /vagrant $GOPATH/src/github.com/alibaba/pouch
+        cd $GOPATH/src/github.com/alibaba/pouch && make
```
Comment (fuweid) [inline Vagrantfile]: without install?

File: VAGRANT.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+# Start a nginx container with 80
+pouch run -d --name nginx -p 80:80 nginx
+curl http://localhost
+```
+
+## Build pouch with vagrant
+
+```bash
+export POUCH_BUILD=true # set POUCH_BUILD=true on Windows
```
Comment (fuweid) [inline VAGRANT.md]: you means the `export xx=yy` in the ubuntu on window?

Comment (manlge) [reply VAGRANT.md]: On Windows export POUCH_BUILD=true doesn't,

File: Vagrantfile (hunk 2 above)
Comment (manlge) [reply Vagrantfile]: OK, I'll change to go1.9.1 next commit.

        GO_VERSION=1.9.1
        GOPATH=/go
        apt-get install -y --no-install-recommends build-essential
        wget --progress=bar:force:noscroll https://dl.google.com/go/go$GO_VERSION.linux-amd64.tar.gz -O /tmp/go$GO_VERSION.linux-amd64.tar.gz
        tar xf /tmp/go$GO_VERSION.linux-amd64.tar.gz -C /opt/
        echo "export GOROOT=/opt/go" >> ~/.bashrc
        echo "export GOPATH=$GOPATH" >> ~/.bashrc
        cd /usr/bin && find /opt/go/bin -type f | xargs -n1 ln -f -s

**Pull request merged**